### ISSUE TITLE
Check dependency updates recursively?

### ISSUE THREAD

**Issue by Mygod** [ben-manes/gradle-versions-plugin#282]
For example, `androidx.preference:preference:1.1.0-alpha02` depends on `androidx.fragment:fragment:1.1.0-alpha02`, but there exists update `androidx.fragment:fragment:1.1.0-alpha03`.

**Comment by ben-manes** [comment]
This is called a transitive dependency and could be done as a task property. Currently we don’t resolve those,
https://github.com/ben-manes/gradle-versions-plugin/blob/ae5154ae66b17742ce82d0549249cd3e32fe7ce6/src/main/groovy/com/github/benmanes/gradle/versions/updates/Resolver.groovy#L149-L151

Usually you only manage explicitly declared dependencies, so it hasn’t been an issue or asked for.

**Comment by Mygod** [comment]
Having this feature still sounds like a good idea though?

**Comment by ben-manes** [comment]
I don’t think it will useful for most people, but it’s not a bad idea and some might prefer it.

**Comment by grimreaper** [comment]
It can be useful to understand if my dependencies don't update *their* deps. I don't think I'd action it  directly but might be useful to understand if some feature I want isn't ready or a dep is lax about security.

**Comment by laxika** [comment]
If you have a lot of in-house dependencies (working on a huge project for example) this could be very useful. You can easily ask other teams to up their dependencies.

**Comment by santoch** [comment]
This feature is extremely useful for finding and diagnosing "diamond" problems, ie, when you depend on 2 projects and they depend on (different versions) of another project.

**Comment by fluidsonic** [comment]
Also useful for us. We use various own libraries and Gradle plugins whose dependencies are never checked when we run `dependencyUpdates` in one of the projects which use these libraries and plugins.
We don't want to go through all of them manually all the time just to check for dependency updates :)
